tests/static-check/run_checks.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+set -x
+
+n_procs="$(getconf _NPROCESSORS_ONLN)"
+
+function check_with_gcc() {
+  make clean
+  ./autogen.sh --enable-debug CC=gcc
```

Comment:
@craigcomstock This one is outdated now, compared to https://github.com/cfengine/core/pull/5706/files

I think they should be the same and we should work towards how we discussed it to be (this test does not rely on autotools - a separate step beforehand should prepare a ready-to-use source tarball where we run configure).

Reply:
yeah, will fix.